src/main/flight/pid.c

```diff
@@ -664,15 +669,24 @@ void pidInitConfig(const pidProfile_t *pidProfile)
         thrustLinearizationB = (1.0f - thrustLinearization) / (2.0f * thrustLinearization);
     }
 #endif    
-#if defined(USE_D_CUT)
-    dtermCutPercent = pidProfile->dterm_cut_percent / 100.0f;
-    dtermCutPercentInv = 1.0f - dtermCutPercent;
-    dtermCutRangeHz = pidProfile->dterm_cut_range_hz;
-    dtermCutLowpassHz = pidProfile->dterm_cut_lowpass_hz;
-    dtermCutGain = pidProfile->dterm_cut_gain * dtermCutPercent * D_CUT_GAIN_FACTOR / dtermCutLowpassHz;
+#if defined(USE_D_MIN)
+    if ((pidProfile->d_min_roll > 0) && (pidProfile->d_min_roll < pidProfile->pid[FD_ROLL].D)){
+        dMinPercentRoll = (pidProfile->d_min_roll) / (float)(pidProfile->pid[FD_ROLL].D);
+    } else {
+    dMinPercentRoll = 0;
```

Comment:
needs to be indented. same with L681.. alternatively, there must be a cleaner way to write this.. perhaps:

``` c
// dMinPercentRoll = 0; // should already FAST_RAM_ZERO_INIT to 0
// dMinPercentRoll = 0 / 20
// -> 0 (when d_min_roll is 0 e.g. by init, dMinPercentRoll will always be 0.)

dMinPercentRoll = pidProfile->d_min_roll / (float)pidProfile->pid[FD_ROLL].D;
dMinPercentPitch = pidProfile->d_min_pitch / (float)pidProfile->pid[FD_PITCH].D;
```

some unit tests could help assert this.

Reply:
Yes I don't like this code at all, I am hoping there is a better way to do it.

Reply:
Thanks to eTracer the code is now much better, think it addresses this concern so will mark as resolved.